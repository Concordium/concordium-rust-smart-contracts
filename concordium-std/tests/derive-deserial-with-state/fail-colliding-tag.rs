//! Ensure `derive(DeserialWithState)` fails when two 'tag' attributes are
//! colliding.
use concordium_std::*;

#[derive(DeserialWithState)]
#[concordium(repr(u8))]
#[concordium(state_parameter = "S")]
enum MyTaggedEnum<S> {
    #[concordium(tag = 42)]
    One(StateMap<u32, String, S>),
    #[concordium(tag = 42)]
    Two,
}

fn main() {}
