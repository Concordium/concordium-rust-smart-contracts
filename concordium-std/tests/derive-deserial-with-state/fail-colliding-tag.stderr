error: The tag of 'Two' collides with the tag of 'One'.
  --> tests/derive-deserial-with-state/fail-colliding-tag.rs:11:18
   |
11 |     #[concordium(tag = 42)]
   |                  ^^^

error: The tag of 'One' collides with the tag of 'Two'.
 --> tests/derive-deserial-with-state/fail-colliding-tag.rs:9:18
  |
9 |     #[concordium(tag = 42)]
  |                  ^^^
